packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/devices.js

```diff
@@ -472,7 +914,97 @@ describe('plugin-devices', () => {
       });
     });
 
-    describe('#unregister()', () => {});
-    describe('#waitForRegistration()', () => {});
+    describe('#unregister()', () => {
+      describe('when the device is registered', () => {
+        beforeEach('register the device', () => devices.register());
+
+        describe('when the unregistration request is sent', () => {
+          let url;
+
+          beforeEach('setup the \'request()\' stub', () => {
+            devices.request = sinon.stub().resolves();
+            url = devices.url;
+          });
+
+          it('should use the device\'s url property', () => {
+            assert.isTrue(devices.registered);
+
+            return devices.unregister()
+              .then(() =>
+                assert.calledWith(devices.request, sinon.match.has(
+                  'uri', url
+                )));
+          });
+
+          it('should send a DELETE request', () => {
+            assert.isTrue(devices.registered);
+
+            return devices.unregister()
+              .then(() =>
+                assert.calledWith(devices.request, sinon.match.has(
+                  'method', 'DELETE'
+                )));
+          });
+        });
+
+        describe('when the device unregistration request is successful', () => {
+          it('should clear the device properties', () => devices.unregister()
+            .then(() => assert.isUndefined(devices.url)));
+        });
+
+        describe('when the device unregistration request fails', () => {
+          beforeEach('setup the \'request\' stub', () => {
+            devices.request = sinon.stub().rejects(
+              new WebexHttpError({
+                statusCode: 404,
+                options: {
+                  url: devices.url,
+                  headers: {
+                    trackingId: 'tid'
+                  }
+                }
+              })
+            );
+          });
+
+          it('returns a rejected promise',
+            () => assert.isRejected(devices.unregister()));
+        });
+      });
+
+      describe('when the device is not registered', () => {
+        it('should return a resolved promise',
+          () => assert.isFulfilled(devices.unregister()));
+      });
+    });
+
+    describe('#waitForRegistration()', () => {
+      describe('when the device is registered', () => {
+        beforeEach('register the device', () => devices.register());
+
+        it('should return a resolved promise',
+          () => assert.isFulfilled(devices.waitForRegistration()));
+      });
+
+      describe('when the device is not registered', () => {
+        describe('when the device registers', () => {
+          beforeEach('trigger a pending device registration', () => {
+            devices.register();
```

Comment:
This is another potential race condition. Let's move it into the it and Promise.all with `waitForRegistration`. Something like:

```js
return Promise.all([devices.register(), devices.waitForRegistration()])
  .then(() => assert.isTrue(devices.registered));
```

Reply:
Moved these to a `Promise.all()`.